lib/matplotlib/tests/test_labeled_data_unpacking.py

```diff
@@ -0,0 +1,493 @@
+                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: None")
+        assert_equal(func(None, x="x", y="y"),
+                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: None")
+        assert_equal(func(None, "x", "y", label=""),
+                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: ")
+        assert_equal(func(None, "x", "y", label="text"),
+                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: text")
+        assert_equal(func(None, x="x", y="y", label=""),
+                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: ")
+        assert_equal(func(None, x="x", y="y", label="text"),
+                     "x: ['x'], y: ['y'], ls: x, w: xyz, label: text")
+
+
+def test_function_call_with_dict_data():
+    """Test with dict data -> label comes from the value of 'x' parameter """
+    data = {"a": [1, 2], "b": [8, 9], "w": "NOT"}
+    for func in all_funcs:
+        assert_equal(func(None, "a", "b", data=data),
+                     "x: [1, 2], y: [8, 9], ls: x, w: xyz, label: b")
+        assert_equal(func(None, x="a", y="b", data=data),
+                     "x: [1, 2], y: [8, 9], ls: x, w: xyz, label: b")
+        assert_equal(func(None, "a", "b", label="", data=data),
+                     "x: [1, 2], y: [8, 9], ls: x, w: xyz, label: ")
+        assert_equal(func(None, "a", "b", label="text", data=data),
+                     "x: [1, 2], y: [8, 9], ls: x, w: xyz, label: text")
+        assert_equal(func(None, x="a", y="b", label="", data=data),
+                     "x: [1, 2], y: [8, 9], ls: x, w: xyz, label: ")
+        assert_equal(func(None, x="a", y="b", label="text", data=data),
+                     "x: [1, 2], y: [8, 9], ls: x, w: xyz, label: text")
+
+
+def test_function_call_with_dict_data_not_in_data():
+    "test for the case that one var is not in data -> half replaces, half kept"
+    data = {"a": [1, 2], "w": "NOT"}
+    for func in all_funcs:
+        assert_equal(func(None, "a", "b", data=data),
+                     "x: [1, 2], y: ['b'], ls: x, w: xyz, label: b")
+        assert_equal(func(None, x="a", y="b", data=data),
+                     "x: [1, 2], y: ['b'], ls: x, w: xyz, label: b")
+        assert_equal(func(None, "a", "b", label="", data=data),
+                     "x: [1, 2], y: ['b'], ls: x, w: xyz, label: ")
+        assert_equal(func(None, "a", "b", label="text", data=data),
+                     "x: [1, 2], y: ['b'], ls: x, w: xyz, label: text")
+        assert_equal(func(None, x="a", y="b", label="", data=data),
+                     "x: [1, 2], y: ['b'], ls: x, w: xyz, label: ")
+        assert_equal(func(None, x="a", y="b", label="text", data=data),
+                     "x: [1, 2], y: ['b'], ls: x, w: xyz, label: text")
+
+
+def test_function_call_with_pandas_data():
+    """test with pandas dataframe -> label comes from data["col"].name """
+    try:
+        import pandas as pd
+    except ImportError:
+        raise SkipTest("Pandas not installed")
```

Comment:
Now available for Travis as:  `pip install --no-index --trusted-host travis-wheels.scikit-image.org --find-links=http://travis-wheels.scikit-image.org pandas` for 2.6, 2.7, 3.3, and 3.4.